src/index.ts

```diff
@@ -7,24 +7,103 @@ const points = new Array<{ x: number, y: number, segmentLength: number }>();
-const output = fft.createComplexArray() as number[];
+let fftSize = 4096;
+let fft = new FFT(fftSize);
+let input = fft.createComplexArray() as number[];
+let output = fft.createComplexArray() as number[];
 const components = new Array<{ frequency: number, magnitude: number, phase: number }>();
 const lines = new Array<{ x: number, y: number }>();
 let parameter = 0;
 let complexity = 0;
 let circles = false;
 let hasCapture = false;
 
+const parameterSlider = document.getElementById('parameter-slider') as HTMLInputElement;
+parameterSlider.oninput = function () {
+    parameter = parameterSlider.valueAsNumber;
+    redraw();
+};
+const complexityNumber = document.getElementById('complexity-number') as HTMLInputElement;
+complexityNumber.oninput = function () {
+    complexity = complexityNumber.valueAsNumber;
+    redraw();
+};
+const complexityCircles = document.getElementById('complexity-circles-check') as HTMLInputElement;
+complexityCircles.oninput = function () {
+    circles = complexityCircles.checked;
+    redraw();
+};
+
 function updateCanvasSize() {
     canvas.width = window.devicePixelRatio * canvas.clientWidth;
     canvas.height = window.devicePixelRatio * canvas.clientHeight;
 }
 
+function loadLocation() { //Inspiration from https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/21152762#21152762 (qd's not stored)
+    if (window.location.search)
+        window.location.search.substr(1).split(`&`)
+            .forEach(item => {
+                let [k, v] = item.split(`=`);
+                if (v != null) { //Restriction to valued keys
+                    v = v && decodeURIComponent(v);
+                    switch (true) {
+                        case (k === 'pt'):
+                            let [x, y] = v.split(';');
+                            if (x != null && y != null)
+                                addPoint(Number(x), Number(y), false);
+                            break;
+
+                        case (k === 'range'):
+                            parameterSlider.value = v;
+                            break;
+
+                        case (k === 'circles'):
+                            complexityCircles.checked = Boolean(Number(v));
+                            break;
+
+                        case (k === 'complexity'):
+                            complexityNumber.value = v;
+                            break;
+
+                        case (k === 'fftsize'):
+                            fftSize = Number(v);
+                            fft = new FFT(fftSize);
+                            input = fft.createComplexArray() as number[];
+                            output = fft.createComplexArray() as number[];
+                            break;
+
+                        default:
+                            break;
+                    }
+                }
+            })
+}
+
+function setLocation() {
+    if (points.length > 0 && 'URLSearchParams' in window) {
```

Comment:
If we intend to support browsers where `'URLSearchParams' in window` is false, we should move this check to control the visibility of the button so that we don't show a button that does nothing in those browsers.

Reply:
Sure. We might aswell find another solution that works on all (oldest browers), not relying on `URLSearchParams`.

Reply:
7a568d0 removed the string check.